src/main/java/com/royalrangers/service/UserService.java

```diff
@@ -154,5 +159,101 @@ public UserAgeGroup determineUserAgeGroup(int userAge) {
         return listUsersBeanToApprove;
     }
 
+    public boolean hasRole(String role) {
+        Collection<GrantedAuthority> authorities = (Collection<GrantedAuthority>)
+                SecurityContextHolder.getContext().getAuthentication().getAuthorities();
+        boolean hasRole = false;
+
+        for (GrantedAuthority authority : authorities) {
+            hasRole = authority.getAuthority().equals(role);
+            if (hasRole) {
+                break;
+            }
+        }
+        return hasRole;
+    }
+
+    public String getLoggedUserEmail() {
+        return SecurityContextHolder.getContext().getAuthentication().getName();
+    }
+
+    public void updateUserByEmail(String email, UserBean update) {
+        if(!hasRole("ROLE_ADMIN")) {
+            shrinkProperties(update);
+        }
+        User user = userRepository.findByEmail(email);
+        updateUser(user, update);
+    }
+
+    private void shrinkProperties(UserBean update) {
```

Comment:
It's ugly method to limit user allowed changes, based on old "null check" update